nomad/structs/structs.go

```diff
@@ -5145,6 +5153,38 @@ func (u *UpdateStrategy) Rolling() bool {
 	return u.Stagger > 0 && u.MaxParallel > 0
 }
 
+func (u *UpdateStrategy) Canonicalize() {
```

Comment:
I worry that having this new functionality will mean users upgrading will see their job specification change between versions. I wonder why we need this, when the canonicalization should be handled by before it reaches the RPC.

I took a look at the use of `structs.DefaultUpdateStrategy` and it seems this only currently gets used by mocks and tests, which suggests we currently rely on the API default instantiation as mentioned above.